hedera-node/hedera-app-spi/src/main/java/com/hedera/node/app/spi/key/KeyUtils.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.spi.key;
+
+import com.hedera.hapi.node.base.ContractID;
+import com.hedera.hapi.node.base.Key;
+import com.hedera.hapi.node.base.Key.KeyOneOfType;
+import com.hedera.hapi.node.base.KeyList;
+import com.hedera.hapi.node.base.ThresholdKey;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+
+public class KeyUtils {
+    public static final int ED25519_BYTE_LENGTH = 32;
+    private static final byte ODD_PARITY = (byte) 0x03;
+    private static final byte EVEN_PARITY = (byte) 0x02;
+    public static final int ECDSA_SECP256K1_COMPRESSED_KEY_LENGTH = 33;
+    /**
+     * Checks if the given key is empty.
+     * For a KeyList type checks if the list is empty.
+     * For a ThresholdKey type checks if the list is empty.
+     * For an Ed25519 or EcdsaSecp256k1 type checks if there are zero bytes.
+     * TODO: This method need to be updated for other key types.
+     * @param pbjKey the key to check
+     * @return true if the key is empty, false otherwise
+     */
+    public static boolean isEmpty(final Key pbjKey) {
+        final var key = pbjKey.key();
+        if (key != null && key.kind().equals(KeyOneOfType.UNSET)) {
+            return true;
+        }
+        if (pbjKey.hasKeyList()) {
+            return ((KeyList) key.value()).hasKeys()
+                    && ((KeyList) key.value()).keys().isEmpty();
+        } else if (pbjKey.hasThresholdKey()) {
+            return ((ThresholdKey) key.value()).hasKeys()
+                    && ((ThresholdKey) key.value()).keys().keys().isEmpty();
+        } else if (pbjKey.hasEd25519()) {
+            return ((Bytes) key.value()).length() == 0;
+        } else if (pbjKey.hasEcdsaSecp256k1()) {
+            return ((Bytes) key.value()).length() == 0;
+        } else if (pbjKey.hasDelegatableContractId()) {
+            return ((ContractID) key.value()).contractNum() == 0;
+        } else if (pbjKey.hasContractID()) {
+            return ((ContractID) key.value()).contractNum() == 0;
+        }
+        // ECDSA_384 and RSA_3072 are not supported yet
+        return true;
+    }
+
+    public static boolean isValid(final Key pbjKey) {
+        if (isEmpty(pbjKey)) {
+            return false;
+        }
+        final var key = pbjKey.key();
+        if (key.kind().equals(KeyOneOfType.UNSET)) {
```

Comment:
Swap to avoid potential NPE?
```suggestion
        if (KeyOneOfType.UNSET.equals(key.kind()) {
```

Reply:
I don't think its possible since PBJ generates it 🙂 . If nothing is set `UNSET` should be present. But I don't mind making this change